### PR TITLE
Update docs for new APT repos

### PR DESCRIPTION
**Pull request by fheinecke**
Update the APT install command to point to the new APT repos as a part of RFD 0058.

Related PR: https://github.com/gravitational/teleport/pull/10746

This is my first doc PR so I'm not sure that I did everything right. Do I need to run a static site generator of some kind? Also, is there anything special I need to do because I added a variable to `config.json`?

### CODE REVIEW

File: docs/pages/includes/install-linux.mdx [hunk 1]
```diff
@@ -5,8 +5,14 @@
         # Download Teleport's PGP public key
         $ sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
           -o /usr/share/keyrings/teleport-archive-keyring.asc
-        # Add the Teleport APT repository
-        $ echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] https://deb.releases.teleport.dev/ stable main" \
+        # Source variables about OS version
+        $ source /etc/os-release
+        # Add the Teleport APT repository for v(=teleport.major_version=). You'll need to update this
+        # file for each major (breaking) release of Teleport.
```
Comment (r0mant) [inline docs/pages/includes/install-linux.mdx:11]:
```suggestion
        # file for each major release of Teleport.
```

File: docs/pages/includes/install-linux.mdx [hunk 2]
```diff
@@ -5,8 +5,14 @@
         # Download Teleport's PGP public key
         $ sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
           -o /usr/share/keyrings/teleport-archive-keyring.asc
-        # Add the Teleport APT repository
-        $ echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] https://deb.releases.teleport.dev/ stable main" \
```
Comment (r0mant) [inline docs/pages/includes/install-linux.mdx:9]: I wonder if we should keep the older repos in a separate tab like "Debian/Ubuntu (Legacy)" or something like this, along with a note explaining that these are the old repos, benefits of the new ones and saying the old ones will be supported for foreseeable future.

@fheinecke @ptgott Thoughts?

File: docs/pages/includes/install-linux.mdx [hunk 3]
```diff
@@ -63,6 +69,20 @@
         ```
   </TabItem>
 
+  <TabItem label="Debian/Ubuntu Legacy (DEB)">
+        ```code
```
Comment (r0mant) [inline docs/pages/includes/install-linux.mdx:73]: Nit: Add a note on this tab explaining deficiencies of these repos and that we recommend using the new ones.

**Review by r0mant — APPROVED** [review]

**Comment by fheinecke** [comment]
As a note this still points to the public GPG key in the old APT repos. I'm working on a PR that will update the repo build tool to have this functionality but it likely will not be merged until after this one.

File: docs/pages/includes/install-linux.mdx (hunk 2 above)
Comment (ptgott) [reply docs/pages/includes/install-linux.mdx:9]: The "Debian/Ubuntu Legacy (DEB)" tab in the current version works for me!

File: docs/pages/includes/install-linux.mdx [hunk 4]
```diff
@@ -5,8 +5,14 @@
         # Download Teleport's PGP public key
         $ sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
           -o /usr/share/keyrings/teleport-archive-keyring.asc
-        # Add the Teleport APT repository
-        $ echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] https://deb.releases.teleport.dev/ stable main" \
+        # Source variables about OS version
+        $ source /etc/os-release
+        # Add the Teleport APT repository for v(=teleport.major_version=). You'll need to update this
+        # file for each major release of Teleport.
+        # Note: if using a fork of Debian or Ubuntu you may need to use '$ID_LIKE'
```
Comment (ptgott) [inline docs/pages/includes/install-linux.mdx:12]: > you may need to use

How can a reader using a fork of Debian/Ubuntu tell whether they will or won't need to use `$ID_LIKE` and the codename their distro was forked from?

File: docs/pages/installation.mdx [hunk 5]
```diff
@@ -40,6 +40,20 @@ All installations include `teleport`, `tsh`, `tctl`, and `tbot`.
...
 (!docs/pages/includes/install-linux.mdx!)
 
+<Details 
+title="Using APT for versions prior to Teleport 10?" 
+scopeOnly={false}
+>
+
+If you've previously installed Teleport via the APT
+repo at `https://deb.releases.teleport.dev/`, you can upgrade by
+re-running the Debian/Ubuntu (DEB) install instructions below.
```
Comment (ptgott) [inline docs/pages/installation.mdx:50]:
```suggestion
re-running the "Debian/Ubuntu (DEB)" install instructions  above.
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**